tests/monotouch-test/Network/NWPathMonitorTest.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using UIKit;
+using Network;
+using CoreFoundation;
+using Foundation;
+
+using NUnit.Framework;
+
+
+namespace monotouchtest.Network {
+    [TestFixture]
+    [Preserve(AllMembers = true)]
+    public class NWPathMonitorTest {
+
+        NWPathMonitor monitor;
+
+        [TestFixtureSetUp]
+        public void Init()
+        {
+            TestRuntime.AssertXcodeVersion (10, 0);
+            monitor = new NWPathMonitor ();
+        }
+
+        [Test]
+        public void StatusPropertyTest ()
+        {
+            Assert.That (monitor.currentPath, Is.Null);
+
+            NWPath finalPath = null;
+            bool isPathUpdated = false;
+            
+            TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () =>
+            {
+
+                monitor.SetUpdatedSnapshotHandler ((path) =>
+                {
+                    if (path != null) {
+                        finalPath = monitor.currentPath;
+                        isPathUpdated = true;
+                    }
+
+                });
+
+                var q = new DispatchQueue (label: "monitor");
+                monitor.SetQueue (q);
+                monitor.Start ();
+
+            }, () => isPathUpdated);
+
+            Assert.That (finalPath, Is.Not.Null);
```

Comment:
when a test has more than one `Assert` call then use a label/message. It can be any string (e.g. "1", "2") as long as it's unique.

The reason is that, sometimes, the debugging symbols are missing and/or the stack trace is off - and it can be hard to determine what failed from logs. Adding a label solves this :)

Reply:
I'm not familiar with testing, but what do the `[TestFixture]` and `[Test]` attributes do/how do they work?

Reply:
https://github.com/nunit/docs/wiki/Attributes :)